src/base/bittorrent/sessionimpl.h

```diff
@@ -824,8 +824,7 @@ namespace BitTorrent
         bool m_isPortMappingEnabled = false;
         QHash<quint16, std::vector<lt::port_mapping_t>> m_mappedPorts;
 
-        QTimer *m_wakeupCheckTimer = nullptr;
-        std::chrono::steady_clock::time_point m_wakeupCheckTimestamp;
+        QElapsedTimer m_wakeupCheckTimestamp;
```

Comment:
Wouldn't it make sense to make it local as well (like m_wakeupCheckTimer)?

Reply:
QElapsedTimer don't have a constructor that can take `QObject parent` so it would become 'untracked' and I don't think that is good.

Reply:
>Use proper data type for elapsed time

Seems `QElapsedTimer` is just wrapper around `std::chrono::steady_clock::time_point`.

Reply:
>Seems QElapsedTimer is just wrapper around std::chrono::steady_clock::time_point.

That is how it is implemented.